library/core/src/num/f32.rs

```diff
@@ -1424,9 +1424,16 @@ impl f32 {
     /// ];
     ///
     /// bois.sort_by(|a, b| a.weight.total_cmp(&b.weight));
-    /// # assert!(bois.into_iter().map(|b| b.weight)
-    /// #     .zip([-5.0, 0.1, 10.0, 99.0, f32::INFINITY, f32::NAN].iter())
-    /// #     .all(|(a, b)| a.to_bits() == b.to_bits()))
+    ///
+    /// if f32::NAN.is_sign_negative() {
```

Comment:
```suggestion
    //// // `f32::NAN` could be positive or negative, which will affect the sort order.
    /// if f32::NAN.is_sign_negative() {
```

Reply:
Applied (manually) since f64.fs also needs the change.